components/dashboard/src/hooks/use-user-loader.ts

```diff
@@ -41,14 +40,18 @@ export const useUserLoader = () => {
         retryDelay: (attemptIndex) => Math.min(1000 * 2 ** attemptIndex, 10000),
         cacheTime: 1000 * 60 * 60 * 1, // 1 hour
         staleTime: 1000 * 60 * 60 * 1, // 1 hour
-        onSuccess: (loadedUser) => {
-            setUser(loadedUser);
-            refreshSearchData();
```

Comment:
@svenefftinge, maybe you recall why this was introduced? 
We've noticed `getSuggestedContextURLs` was called 3x on load of `/new`. Explicitly updating on `setUser` seems to be less obvious, but maybe there was some reason to introduce this? git history shows that there were many changes related to the transition from contexts holding/updating data to usage of query, and I wonder if this is a left-over.

Reply:
Does RepositoryFinder depend on any user properties that it should be retriggered? if not it should not be here? It looks rather strange way of doing, we would provide user state as a dependency to RepositoryFinder if we want to do it properly? cc @selfcontained

Reply:
Good question! 🤷🏻‍♂️ 
Maybe if you're pasting in a repository URL of a SCM provider you did not authorized with, and the context resolution fails, which will render the `Authorize with provider` action? But even if that's the case, and we'd call `setUser` on a success event, we should try to refresh search data from there, and not do it on every update, including the initial one. 
I'll try to reproduce, and see how that's behaving.